### ISSUE TITLE
Make an additional flag for installing CRDs

### ISSUE THREAD

**Issue by gecube** [minio/operator#1915]
Hello!

Other projects provides a flag like `installCRDs` for better management what to install - the operator or the bundle operator + CRDs.
The good example is cert-manager project: https://github.com/cert-manager/cert-manager/blob/529d8a74a679a11757ed7a30ea60d342acbdb484/deploy/charts/cert-manager/values.yaml#L53

Another possible option is to move CRDs to the separate helm chart like it was did in linkerd: https://artifacthub.io/packages/helm/linkerd2/linkerd-crds

It allows more flexible approach and workarounds some Helm bugs regarding CRDs installation. So I am kindly asking to consider these options and implement them (one of two).

**Comment by alistarle** [comment]
I made a proposition of a simple and not breaking implementation.

Moving the CRDs definitions in a "crds" directory of the chart will allow to use the built-in crd support of helm: https://helm.sh/docs/chart_best_practices/custom_resource_definitions/ but it doesn't support upgrade or deletion.

Separate chart is a good idea but it can be breaking, or we should use helm subchart installed by default but even in that case I don't know how helm will handle upgrade to move from a monolitic chart to a main chart + subchart for CRDs.

**Comment by pjuarezd** [comment]
> I made a proposition of a simple and not breaking implementation.
> 
> Moving the CRDs definitions in a "crds" directory of the chart will allow to use the built-in crd support of helm: https://helm.sh/docs/chart_best_practices/custom_resource_definitions/ but it doesn't support upgrade or deletion.
> 
> Separate chart is a good idea but it can be breaking, or we should use helm subchart installed by default but even in that case I don't know how helm will handle upgrade to move from a monolitic chart to a main chart + subchart for CRDs.

Right, we came to a similar conclusion, please check comment here https://github.com/minio/operator/issues/1553#issuecomment-1650661259

**Comment by allanrogerr** [comment]
We will discuss this internally and get back to you.

**Comment by fredgate** [comment]
Putting CustomResourceDefinitions in the `crds/` of the chart instead of `templates/`, avoid that the CRDs (and by cascade all the CR of these types) be deleted when the release of this chart is uninstalled.
